doc/dev/installation.rst

```diff
@@ -63,13 +63,117 @@ from within the root of the running container.
+        sudo apt install clang lld ccache libomp-dev ninja-build make cmake 
+
+        # Clone the Catalyst repository  
+        git clone --recurse-submodules --shallow-submodules https://github.com/PennyLaneAI/catalyst.git
+
+        # Install specific requirements for Catalyst
+        cd catalyst
+        pip install -r requirements.txt
+
+        # Build Catalyst
+        make all
+
+
+   .. group-tab:: macOS
+
+      .. code-block:: console
+
+        # Install XCode Command Line Tools and common requirements
+        xcode-select --install
+        pip install cmake ninja
+        brew install libomp
+
+        # Clone the Catalyst repository  
+        git clone --recurse-submodules --shallow-submodules https://github.com/PennyLaneAI/catalyst.git
+
+        # Install specific requirements for Catalyst
+        cd catalyst
+        pip install -r requirements.txt 
+
+        # Build Catalyst
+        make all
+
+
+
+The essential steps should give you the full functionality of Catalyst. There are also some optional steps:
+
+
+.. tabs::
+
+   .. group-tab:: Linux Debian/Ubuntu
+
+      .. code-block:: console
+
+        # If you are building Catalyst components in custom locations, set environment variables. 
+        # This step should not be required if you are simply following the instructions above. 
+        # However, performing this step anyway should not cause any failures. 
+        export PYTHONPATH="$PWD/mlir/build/python_packages/quantum:$PYTHONPATH"
+        export RUNTIME_LIB_DIR="$PWD/runtime/build/lib"
+        export MLIR_LIB_DIR="$PWD/mlir/llvm-project/build/lib"
+        export ENZYME_LIB_DIR="$PWD/mlir/Enzyme/build/Enzyme"
+        export PATH="$PWD/mlir/llvm-project/build/bin:$PWD/mlir/mlir-hlo/mhlo-build/bin:$PWD/mlir/build/bin:$PATH"
+
+        # Optional: test that everything is built properly
+        make test
+
+        # Optional: build and test documentation for Catalyst
+        pip install -r doc/requirements.txt
+        sudo apt install doxygen pandoc
+
+   .. group-tab:: macOS
+
+      .. code-block:: console
+
+        # If you are building Catalyst components in custom locations, set environment variables. 
+        # This step should not be required if you are simply following the instructions above. 
+        # However, performing this step anyway should not cause any failures. 
+        export PYTHONPATH="$PWD/mlir/build/python_packages/quantum:$PYTHONPATH"
+        export RUNTIME_LIB_DIR="$PWD/runtime/build/lib"
+        export MLIR_LIB_DIR="$PWD/mlir/llvm-project/build/lib"
+        export ENZYME_LIB_DIR="$PWD/mlir/Enzyme/build/Enzyme"
+        export PATH="$PWD/mlir/llvm-project/build/bin:$PWD/mlir/mlir-hlo/mhlo-build/bin:$PWD/mlir/build/bin:$PATH"
+
+        # Optional: test that everything is built properly
+        make test
+
+        # Optional: build and test documentation for Catalyst
+        pip install -r doc/requirements.txt
+        brew install doxygen pandoc
```

Comment:
I'm not quite sure these are necessary for the minimal guide. 

- Custom paths seems like something that is better covered in the detailed guide, since a user just copy-pasting the install commands is unlikely to set custom installation paths.

- `make test` could be moved into the section above (if you agree), it's actually a nice idea to double check the installation worked :) 

- Building documentation is also a rather "advanced thing".

Reply:
Agree!

Reply:
> My rationale is that if the user installed in a custom path, then not setting the variables will cause catalyst to not work, so I included them in the minimal guide.

But the essential guide doesn't cover how the user would set custom build directories, so I don't see how that would have happened (note that custom build directories would require you to explicitly set environment variables before running make commands, or manually invoe cmake commands, all pretty advanced stuff).
So I think setting the paths is more likely to confuse users in this case.

Reply:
Double checked the docs and even the detailed guide doesn't tell you how to set custom build directories

Reply:
These are the variables you would have to override when invoking make:
https://github.com/PennyLaneAI/catalyst/blob/a723cd16fc961f0609bf489eb6676c2b3b9f351d/Makefile#L10-L15

Reply:
Good point, I will remove the variables from essentials.